org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/core/ReplModule.java

```diff
@@ -25,9 +26,11 @@
 import com.google.inject.multibindings.MapBinder;
 
 /**
- * Client library bindings.
+ * This class binds the core classes. It is intended to be subclassed by client implementations.
+ * These subclasses should bind their implementations of {@link IRepl}, {@link IMessageHook} and
+ * {@link IResultHook}.
  */
-public class ReplModule extends SpoofaxModule {
+public abstract class ReplModule extends SpoofaxModule {
```

Comment:
Dummies can print warnings, just like the Spoofax dummies do. Example: https://github.com/metaborg/spoofax/blob/692ef4677d4a1b70494b0fc12f0c2b013408d247/org.metaborg.core/src/main/java/org/metaborg/core/project/DummyProjectService.java